src/store/actions.ts

```diff
@@ -0,0 +1,370 @@
+          channel,
+        ),
+      });
+    } else {
+      commit('postMessage', {
+        message: new Message(`${username} has already been ignored`, channel),
+      });
+    }
+  },
+  sendMessageRaw({ state }) {
+  },
+  userKicked(
+    { state, commit, dispatch },
+    {
+      params,
+    }: {
+      params: IrcKickEventArgs;
+    },
+  ) {
+    const username = parseUsername(params.nick);
+    if (username === state.currentUser.username) {
+      state.banned[params.channel] = consts.BAN_STATUS_BANNED;
+      commit('postMessage', {
+        message: new Message(
+          `You have been kicked from chat${params.message ? ` - ${params.message}` : ''}`,
+        ),
+      });
+    } else {
+      commit('removeUser', { username });
+    }
+  },
+  onMessageReceived(
+    { state, commit, dispatch },
+    {
+      message, nick, tags, time, type, target,
+    }: IrcMessageEventArgs,
+  ) {
+    const username = parseUsername(nick);
+    commit('postMessage', {
+      message: new Message(message, target, state.connectedUsers[username], type === 'action', time),
+    });
+  },
+  onUserQuit(
+    { state, commit, dispatch },
+    {
+      message,
+    }: {
+      message: any;
+    },
+  ) {
+    commit('removeUser', { username: message });
+  },
+  onModeMessageRecieved({ state, commit, dispatch }, event: IrcModeEventArgs) {
+    for (const mode of event.modes) {
+      let mute = false;
+      if (mode.param && mode.param.startsWith('m;')) {
+        mute = true;
+        mode.param = mode.param.substr(2);
+      }
+      // Check if message has been either banned or muted, if so disable input and notify in chat
+      if (mode.mode === '+b') {
+        const maskParts = mode.param.match(/(~q:)?(.+)!.+/)!;
+        if (state.nick.match(new RegExp(maskParts[2].replace('*', '.+').replace('^', '\\^')))) {
+          state.banned[event.target] = mute ? consts.BAN_STATUS_QUIET : consts.BAN_STATUS_BANNED;
+          if (maskParts[1]) {
+            commit('postMessage', {
+              message: new Message('You are no longer allowed to post in chat', event.target),
+            });
+          } else {
+            commit('postMessage', {
+              message: new Message(
+                `You have been ${mute ? 'muted' : 'banned'} from chat`,
+                event.target,
+              ),
+            });
+          }
+        }
+      } else if (mode.mode === '-b') {
+        const maskUser = mode.param.match(/(?:~q:)?(.+)!.+/)![1];
+        if (state.nick.match(new RegExp(maskUser.replace('*', '.+').replace('^', '\\^')))) {
+          state.banned[event.target] = consts.BAN_STATUS_NORMAL;
+          commit('postMessage', {
+            message: new Message('You are now allowed to post in chat', event.target),
+          });
+        }
+      }
+    }
+  },
+  onRawMessageRecieved(
```

Comment:
You should be able to listen for ERR_CANNOTSENDTOCHAN/ERR_BANNEDFROMCHAN directly - https://github.com/kiwiirc/irc-framework/blob/fba333225948ea38a1aa40c68f7cc6f6f82776a1/src/commands/handlers/generics.js#L137